File: src/TinRoll.Test/Mapper/UserMapperTests.cs
//using System;
//using TinRoll.Data.Entities;
//using TinRoll.Logic.Mapper;
//using TinRoll.Shared;
//using Xunit;

//namespace TinRoll.Test.Mapping
//{
//    public class UserMapperTests
//    {
//        [Fact]
//        public void Test_User_To_UserDto()
//        {
//            //arrange
//            var user = new User
//            {
//                Id = 1,
//                CreatedDate = DateTime.UtcNow,
//                Email = "testEmail1",
//                UpdatedDate = DateTime.UtcNow,
//                UserName = "testUsername"
//            };

//            //act
//            var userDto = UserMapper.ToDto(user);

//            //assert
//            Assert.NotNull(userDto);
//            Assert.Equal(user.Id, userDto.Id);
//            Assert.Equal(user.CreatedDate, userDto.CreatedDate);
//            Assert.Equal(user.Email, userDto.Email);
//            Assert.Equal(user.UpdatedDate, userDto.UpdatedDate);
//            Assert.Equal(user.UserName, userDto.UserName);
//        }

//        [Fact]
//        public void Test_UserDto_To_User()
//        {
//            //arrange
//            var userDto = new UserDto
//            {
//                Id = 1,
//                CreatedDate = DateTime.UtcNow,
//                Email = "testEmail1",
//                UpdatedDate = DateTime.UtcNow,
//                UserName = "testUsername"
//            };

//            //act
//            var user = UserMapper.ToDb(userDto);

//            //assert
//            Assert.NotNull(userDto);
//            Assert.Equal(userDto.Id, user.Id);
//            Assert.Equal(userDto.CreatedDate, user.CreatedDate);
//            Assert.Equal(userDto.Email, user.Email);
//            Assert.Equal(userDto.UpdatedDate, user.UpdatedDate);
//            Assert.Equal(userDto.UserName, user.UserName);
//        }
//    }
//}

File: src/TinRoll.Test/Logic/Mapper/UserMapperTests.fs
module Logic.Mapper.UserMapperTests

open Xunit
open TinRoll.Data
open TinRoll.Logic.Mapper
open System.Collections
open System
open NFluent
open TinRoll.Shared

[<Fact>]
let ``Test User Db to Dto`` () =
    let user = new User (1, "terst@gail.com", "TestUsername", new Generic.List<Question>(), DateTime.MinValue, DateTime.MinValue)
    let dto = UserMapper.ToDto(user)

    Check.That(dto.Id).Equals(user.Id) |> ignore
    Check.That(dto.Email).Equals(user.Email) |> ignore
    Check.That(dto.Username).Equals(user.Username) |> ignore    
    Check.That(dto.CreatedDate).Equals(user.CreatedDate) |> ignore
    Check.That(dto.UpdatedDate).Equals(user.UpdatedDate) |> ignore

[<Fact>]
let ``Test User Dto to Db`` () =
    let dto = new UserDto (Id = 1, Email = "tesrst@gmail.com", Username = "Testusername", Questions = new Generic.List<QuestionDto>(), CreatedDate = DateTime.MinValue, UpdatedDate = DateTime.MinValue)
    let db = UserMapper.ToDb(dto)

    Check.That(db).IsNotNull() |> ignore
    Check.That(db.Id).Equals(dto.Id) |> ignore
    Check.That(db.Email).Equals(dto.Email) |> ignore
    Check.That(db.Username).Equals(dto.Username) |> ignore
    Check.That(db.CreatedDate).Equals(dto.CreatedDate) |> ignore
    Check.That(db.UpdatedDate).Equals(dto.UpdatedDate) |> ignore